backend-lucene/src/main/java/org/hibernate/search/v6poc/backend/lucene/search/predicate/impl/AbstractSpatialWithinCirclePredicateBuilder.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.v6poc.backend.lucene.search.predicate.impl;
+
+import org.hibernate.search.v6poc.backend.spatial.GeoPoint;
+import org.hibernate.search.v6poc.search.predicate.spi.SpatialWithinCirclePredicateBuilder;
+
+
+public abstract class AbstractSpatialWithinCirclePredicateBuilder<T> extends AbstractSearchPredicateBuilder
```

Comment:
Not sure I follow you: if you want to search a polygon in a circle, you will need to define the circle. That's all that is in `AbstractSpatialWithinCirclePredicateBuilder`.

Reply:
I am mainly talking about additional, optional parameters. I can imagine searching for polygons within a circle being similar, but slightly different from search for points within a circle.

In particular, when searching for points within a circle, you will use a distance query, which involves picking some distance computation algorithm (`arc` or `plane` in Elasticsearch).
When searching for polygons within a circle, you will not use a distance query, at least not in Elasticsearch; you will use a `geo_shape` query. And this query does not allow you to pick a distance computation algorithm (maybe because it doesn't use one).

As a result, if we decide to expose a `.distanceAlgorithm(DistanceAlgorithmEnum)` method on the object returned by a call to `circle`, our API will be dodgy: in some cases ("point" fields) that method can be called, in some other cases ("shape" fields) it cannot.

That's just an example though: looking for points within a circle is not exactly the same as looking for shapes within a circle, so I would expect other differences that could translate into different APIs.

Reply:
Well, if need be and these classes ends up being useless, we will be able to remove them.

I kinda liked the symmetry with the other types too.